### DISCUSSION TITLE
Updated Peripheral MicePoke

### DISCUSSION THREAD

**Discussion by Poofjunior** [harp-tech/protocol#107] [Open-ended]
Hi folks,

On the AIND side, were planning to conduct a large scale (50+ units) mouse poke-based experiment, and I am in the process of revising the mouse poke port for easier manufacturability.

The updated design preserves the original port pocket geometry, but adjusts the external geometry to use an [off-the-shelf photointerrupter](https://www.digikey.com/short/z2vvfqwf) instead of chopping up the original into two pieces.

Draft images below:
![image](https://github.com/user-attachments/assets/0713da2b-7818-4cda-bab5-0d645e070f08)
![image](https://github.com/user-attachments/assets/0b00f355-620b-4a25-a73d-077cbea22e32)

Ultimately, the plan is to attach a very small adapter PCB to expose the leads to a connector.

Would folks be interested in versioning this into the [peripheral.micepoke](https://github.com/harp-tech/peripheral.micepoke) project?

**Comment by Poofjunior** [comment]
3D printed fit test looks good:

<p align="center">
  <img alt="" src="https://github.com/user-attachments/assets/6c6368b0-0bfe-4be1-b12b-e03d5978a8a7" width="45%">
&nbsp; &nbsp; &nbsp; &nbsp;
  <img alt="" src="https://github.com/user-attachments/assets/f0d772a0-7bd1-482a-8f94-afd7bd17f05c" width="45%">
</p>
<p align="center">
  <img alt="" src="https://github.com/user-attachments/assets/4c62b587-8cbe-4cb9-a847-536a6a3f0208" width="45%">
&nbsp; &nbsp; &nbsp; &nbsp;
  <img alt="" src="https://github.com/user-attachments/assets/9d9f1a2b-8319-4de5-b65a-2f256d224ee6" width="45%">
</p>

**Comment by artursilva0** [comment]
@Poofjunior Thanks for sharing this! It really makes the production easier. 

Are you planning to add an RJ connector for interfacing with the Harp Behavior? If so, that would keep general compatibility, maybe we could version this one as major. If not, we could just rename this poke and start with version 1.0.

Either way, I'm happy to have this new poke in harp-tech regardless of the versioning used.

**Comment by Poofjunior** [comment]
Thanks for the feedback!

> Are you planning to add an RJ connector for interfacing with the Harp Behavior? If so, that would keep general compatibility, maybe we could version this one as major. If not, we could just rename this poke and start with version 1.0.

To keep it small, I'm probably going to get rid of the RJ45 connector. However, if someone wants to interface something like a behavior board to this device, we can make a custom cable that has the connector on the target PCBA and terminates with RJ45.

Cool! It'd be nice to get this into more hands once it's a bit more mature. I'll keep this thread going with progress updates as they come in.

**Comment by Poofjunior** [comment]
First rev of the board + mechanical port (aluminum) is in! Here are some glamor shots.

The connector is from 3M's Mini Clamp Connector series, and the idea is that the other connector end of the cable could be whatever you want.

It's laughably small, but preserves the original geometry.

![PXL_20250306_011005084~2.jpg](https://github.com/user-attachments/assets/2dee5c77-4648-4a51-af31-ba10429f7adc)

![PXL_20250306_011017762~2.jpg](https://github.com/user-attachments/assets/22f4f2ff-bf9d-44d3-8c4e-2093deab6f25)

![PXL_20250306_011032527~2.jpg](https://github.com/user-attachments/assets/fb5e19e2-7f1b-408c-9598-a2530573e59c)

![PXL_20250306_011102451~2.jpg](https://github.com/user-attachments/assets/55528256-dc3f-473f-8420-e50972499bde)
